src/api/lib/tasks/dev/requests.rake

```diff
@@ -199,5 +199,357 @@ namespace :dev do
+
+        if repository['arch'].is_a?(Array)
+          repository['arch'].each do |architecture|
+            proj_repo.repository_architectures.find_or_initialize_by(architecture: Architecture.find_by(name: architecture))
+          end
+        else
+          proj_repo.repository_architectures.find_or_initialize_by(architecture: Architecture.find_by(name: repository['arch']))
+        end
+        proj_repo.save!
+        project.store(comment: "Added #{proj_repo.name} repository")
+      end
+    end
+    # rubocop:enable Metrics/CyclomaticComplexity
+    # rubocop:enable Metrics/PerceivedComplexity
+
+    # rubocop:disable Metrics/CyclomaticComplexity
+    # rubocop:disable Metrics/PerceivedComplexity
+    def create_users_and_groups_from_meta(request_data:, users:)
+      print_message('creating users and groups from _meta ....')
+      # create users
+      if request_data['person'].present?
+        request_data['person'].each do |person|
+          create_user(login: person['userid'], users: users)
+        end
+      end
+
+      return if request_data['group'].nil?
+
+      if request_data['group'].is_a?(Array)
+        request_data['group'].each do |group|
+          next if Group.find_by(title: group['groupid'])
+
+          create(:group, title: group['groupid'])
+        end
+      else
+        group = Group.find_by(title: request_data['group']['groupid'])
+
+        create(:group, title: request_data['group']['groupid']) if group.nil?
+      end
+    end
+    # rubocop:enable Metrics/CyclomaticComplexity
+    # rubocop:enable Metrics/PerceivedComplexity
+
+    def create_user(login:, users: [])
+      return if User.find_by(login: login).present?
+
+      users << create(:staff_user, login: login)
+    end
+
+    def find_user(login)
+      return if login.nil?
+
+      user = User.find_by_login(login)
+      return user if user
+
+      create(:confirmed_user, login: login)
+    end
+
+    def find_group(title)
+      return if title.nil?
+
+      Group.find_or_create_by!(title: title)
+    end
+
+    def find_project(project_name)
+      return if project_name.nil?
+
+      project = Project.find_by_name(project_name)
+      return project if project
+
+      create(:project, name: project_name)
+    end
+
+    def find_package(project_name, package_name)
+      return if project_name.nil? || package_name.nil?
+
+      package = Package.find_by_project_and_name(project_name, package_name)
+      return package if package
+
+      project = Project.find_by_name(project_name)
+      project = create(:project, name: project_name) if project.nil?
+
+      create(:package, name: package_name, project: project)
+    end
+
+    def clone_prj_configs(config:, comment:, project:)
+      params = ActionController::Parameters.new({ config: config, comment: comment })
+      ProjectConfigurationService::ProjectConfigurationUpdater.new(project, User.session!, params).call
```

Comment:
This is a 🐄 💩 service. It's just doing `project.config.save({user: 'Admin', comment: comment}, config)`

Reply:
Thanks for the suggestion. You are correct. I've updated it